workflow/controller/operator.go

```diff
@@ -1872,7 +1872,7 @@ func (woc *wfOperationCtx) executeContainer(nodeName string, templateScope strin
 	node := woc.wf.GetNodeByName(nodeName)
 	if node == nil {
 		node = woc.initializeExecutableNode(nodeName, wfv1.NodeTypePod, templateScope, tmpl, orgTmpl, opts.boundaryID, wfv1.NodePending)
-	} else if !isResubmitAllowed(tmpl) && !node.Pending() {
+	} else if !node.StartedAt.IsZero() && (!isResubmitAllowed(tmpl) && !node.Pending()) {
```

Comment:
fixed by @sarabala1979 in #3141

Reply:
@sarabala1979 you changed this condition from:

```
!isResubmitAllowed(tmpl) || !node.Pending()
```

to 

```
!isResubmitAllowed(tmpl) && !node.Pending() 
```

and this has fixed the same memoization bug I was trying to fix - was that your intention?